Refactoring for version 181

OndrejM opened the pull request:
Preview: https://docs.payara.fish/v/181refactoring/

No content added, only changed structure and links, plus modified README.adoc and intro pages for previous extended and core docs to make sense (merged core-documentation.adoc and extended-documentation.adoc into payara-server.adoc)

Please don't comment on the actual content unless you find a blatant error that should be fixed ASAP. We need to merge this quickly to avoid merge conflicts.

- top-level Payara Server and Payara Micro documentation (merged core and extended documentation)
- rearranged the general info section to have only 2 root pages
- separate roots for MicroProfile and Ecosystem (was already in 174)
- moved User Guides into a separate section away from technical documentation
- introduced `glassfishVersion` variable to refer to the corresponding GF version (4 or 5)

Still to do in further PRs later:
- create a developer documentation section for provided APIs (Java EE, MicroProfile, Payara-API, Payara Micro API, JCache, Hazelcast, etc.)
- create a separate branch for Payara 5

fturizo commented on documentation/payara-server/payara-server.adoc:
```diff
@@ -0,0 +1,11 @@
+
+Documentation of Payara Server features. Includes full documentation of the features added on top of _GlassFish Server {{ book.glassfishVersion }} Open Source Edition_ as well as features shared with GlassFish Server or their improvements specific to Payara Server.
+
+All documentation for _GlassFish Server {{ book.glassfishVersion }}_ is also valid for Payara Server {{ book.currentVersion }}. 
```
It'd to good to say: " is also valid for Payara Server {{ book.currentVersion }} unless stated otherwise."

OndrejM commented:
@fturizo I added the changes you suggested.

mikecroft commented:
Looks great at first glance! Might be worth clicking through every page just to make sure all the links are correct, then let's get it merged 👍

OndrejM commented:
The links are hopefully OK, I updated them with automatic find&replace across all files, @mikecroft 

I'll wait until Tuesday to give you time to check links and everything, then I'll merge.

michaelranaldo commented on documentation/payara-server/hazelcast/enable-hazelcast.adoc:
```diff
@@ -30,7 +30,7 @@ image::/images/hazelcast/hazelcast-admin-console-enable-hazelcast.png[Payara Ser
 specify whether or not Hazelcast is enabled each time you run it. This
 command is also used to configure Hazelcast, which is covered
-link:/documentation/extended-documentation/hazelcast/configuring-hazelcast-using-asadmin[in the configuration section].
+link:/documentation/payara-server/hazelcast/configuring-hazelcast-using-asadmin[in the configuration section].
```
This link doesn't work. Perhaps unrelated to this change.

OndrejM replied:
This should probably point to [Configuring Hazelcast using Asadmin](https://github.com/payara/Payara-Server-Documentation/blob/9262712902feaf4c2a3dc6f81a75a6804cf14b1e/documentation/payara-server/hazelcast/configuration.adoc#configuring-hazelcast-using-asadmin)

michaelranaldo commented on documentation/payara-server/notification-service/notification-service.adoc:
```diff
@@ -6,8 +6,8 @@ _Since Payara Server 4.1.1.163_
 the
-link:/documentation/extended-documentation/jmx-monitoring-service/jmx-monitoring-service.adoc[JMX Monitoring Service], the link:/documentation/extended-documentation/health-check-service/health-check-service.adoc[HealthCheck service] or the
-link:/documentation/extended-documentation/request-tracing-service/request-tracing-service.adoc[Request Tracing service].
+link:/documentation/payara-server/jmx-monitoring-service/jmx-monitoring-service.adoc[JMX Monitoring Service], the link:/documentation/payara-server/health-check-service/health-check-service.adoc[HealthCheck service] or the
```
The link to HealthCheck service is broken - "/health-check-service.adoc" could be unnecessary.

OndrejM replied:
Good point.
I completely forgot about this README gitbook feature - I'll also fix payara-server.adoc to README.adoc to shorten the link

michaelranaldo commented on documentation/payara-micro/clustering/clustering-with-full-server.adoc:
```diff
@@ -9,5 +9,5 @@ Hazelcast cluster configuration as configured in your Payara Server (e.g. multic
 When Payara Server DAS is connected to the same Hazelcast cluster, the Admin
 console can display the details about Payara Micro instances in the cluster.
-See link:/documentation/extended-documentation/hazelcast/cluster-members.adoc[Management of Hazelcast Cluster Members]
+See link:/documentation/payara-server/hazelcast/cluster-members.adoc[Management of Hazelcast Cluster Members]
```
This link doesn't appear to work.

OndrejM replied:
The page doesn't exist. I only found it in [this revision](https://github.com/payara/Payara-Server-Documentation/blob/fe173eb7a4ac96169e9aaf79298a7b422b8202f4/documentation/extended-documentation/hazelcast/cluster-members.adoc) but in no real version branch.

I think the link should point to [Viewing Hazelcast clusters](https://github.com/payara/Payara-Server-Documentation/blob/master/documentation/extended-documentation/hazelcast/viewing-members.adoc)

OndrejM commented:
Fixed all @michaelranaldo's comments. Also fixed external links, which were prefixed with` link::`, as this is unnecessary in Asciidoc and breaks HTTP links
